lib/CodeGen/CGExprScalar.cpp

```diff
@@ -2319,6 +2319,106 @@ static BinOpInfo createBinOpInfoFromIncDec(const UnaryOperator *E,
   return BinOp;
 }
 
+static Expr *peelOffPointerArithmetic(const BinaryOperator *B) {
```

Comment:
Nit: I guess this function is getting the pointer type for expressions like (a + i) or (i + a), where 'a' is the pointer type and 'i' is the integer offset. Can we add a comment describing this?

Also, maybe we can rename the function to be more clear on the intent? Something like: getPointerTypeFromPointerOffsetExpression or something similar.

Reply:
`peelOffPointerArithmetic` has been adapted from a similar function in `lib/StaticAnalyzer/Core/BugReporterVisitors.cpp`. I wanted to keep the same name to make the association clear and also to help while grep'ping for the function.

Reply:
Added comment describing the function.